### PR TITLE
SAT-38683 - Support any location

### PR DESCRIPTION
**Pull request by lfu**
#### What are the changes introduced in this pull request?
When setting "Any location" on the location dropdown menu, accessing either Remediations or Vulnerabilities menu should work.

#### What are the testing steps for this pull request?
Get an IoP server.
Change the location to "Any location" and visit either the Remediations or Vulnerabilities menu.
Data should be displayed.

## Summary by Sourcery

Allow Insights Cloud requests and tag updates to work when a user selects Any location (nil location).

New Features:
- Support forwarding Insights Cloud UI requests when the session has no location selected (Any location).

Bug Fixes:
- Ensure tag generation works when location is nil by using a wildcard location tag and relaxing host filtering on location.

Tests:
- Add controller tests for forwarding requests with nil and non-nil locations, and a unit test for tag generation with a wildcard location.

### CODE REVIEW

File: app/controllers/insights_cloud/ui_requests_controller.rb [hunk 1]
```diff
@@ -95,7 +95,6 @@ def ensure_org
 
     def ensure_loc
       @location = Location.current
-      return render_message 'Location not found or invalid', :status => 400 unless @location
     end
```
Comment (jeremylenz) [inline app/controllers/insights_cloud/ui_requests_controller.rb]: Should remove this entire method and callback, if we're not going to use it.

**Review by jeremylenz — COMMENTED** [review]
Thanks @lfu!

General question - Why only Location and not Org as well? What would it look like to fix both?

Haven't tested yet but one nit below

File: app/controllers/insights_cloud/ui_requests_controller.rb (hunk 1 above)
Comment (lfu) [reply app/controllers/insights_cloud/ui_requests_controller.rb]: This method is still required for location setting. 
Just it does not reject any location.

Comment (jeremylenz) [reply app/controllers/insights_cloud/ui_requests_controller.rb]: ah, gotcha. Then it should be renamed `find_location` for consistency with similar methods.

**Review by LadislavVasina1 — APPROVED** [review]
ACK, from the QE side, with packit build, Recommendation and Vulnerability pages are not displaying errors anymore, while the Any Location is selected, as they were previously.

Agree with the question raised by Jeremy about the "Any Organization" functionality.

**Review by nofaralfasi — APPROVED** [review]
Tested the changes and verified expected behavior
The PR works as expected and the code looks good.

**Comment by nofaralfasi** [comment]
> Agree with the question raised by Jeremy about the "Any Organization" functionality.

I believe we decided not to implement this for now and to track it separately as its own issue.
cc to @jeremylenz.
